services/worker/src/worker/job_runners/split/duckdb_index.py

```diff
@@ -78,6 +94,76 @@ def check_indexable(feature: FeatureType) -> None:
     return indexable_columns
 
 
+def compute_length_column(
+    parquet_directory: Path,
+    column_name: str,
+    target_df: Optional[pl.DataFrame],
+    dtype: LengthDtype,
+) -> pl.DataFrame:
+    column_class = ListColumn if dtype == "list" else StringColumn
+    df = pl.read_parquet(str(parquet_directory / "*.parquet"), columns=[column_name])
+    lengths_column_name = f"{column_name}.length"
+    lengths_df: pl.DataFrame = column_class.compute_transformed_data(
+        df, column_name, transformed_column_name=lengths_column_name
+    )
+    if target_df is None:
+        return lengths_df.select(pl.col(lengths_column_name))
+
+    target_df.insert_column(target_df.shape[1], lengths_df[lengths_column_name])
+    return target_df
+
+
+def compute_audio_duration_column(
+    parquet_directory: Path,
+    column_name: str,
+    target_df: Optional[pl.DataFrame],
+) -> pl.DataFrame:
+    duration_column_name = f"{column_name}.duration"
+    durations = AudioColumn.compute_transformed_data(parquet_directory, column_name, AudioColumn.get_duration)
+    duration_df = pl.from_dict({duration_column_name: durations})
+    if target_df is None:
+        return duration_df
+    target_df.insert_column(target_df.shape[1], duration_df[duration_column_name])
+    return target_df
+
+
+def compute_image_width_length_column(
+    parquet_directory: Path,
+    column_name: str,
+    target_df: Optional[pl.DataFrame],
+) -> pl.DataFrame:
+    shapes = ImageColumn.compute_transformed_data(parquet_directory, column_name, ImageColumn.get_shape)
+    widths, heights = list(zip(*shapes))
+    width_column_name, height_column_name = f"{column_name}.width", f"{column_name}.height"
+    shapes_df = pl.from_dict({width_column_name: widths, height_column_name: heights})
+    if target_df is None:
+        return shapes_df
+    target_df.insert_column(target_df.shape[1], shapes_df[width_column_name])
+    target_df.insert_column(target_df.shape[1], shapes_df[height_column_name])
+    return target_df
+
+
+def compute_transformed_data(parquet_directory: Path, features: dict[str, Any]) -> Optional[pl.DataFrame]:
+    transformed_df = None
+    for feature_name, feature in features.items():
+        if isinstance(feature, list) or (isinstance(feature, dict) and feature.get("_type") == "Sequence"):
```

Comment:
if `feature` is `Sequence` and `feature.feature` is a dict, then `feature` is a pa.struct()

Checking this should also avoid using `is_list_pa_type` (you already have all the necessary info in `features`)